suites/message/nested.go

```diff
@@ -210,23 +212,126 @@ func TestNestedSends(t *testing.T, factory state.Factories) {
+	//})
+
+	t.Run("fail missing params", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+
+		stage := prepareStage(td, acctDefaultBalance, multisigBalance)
+		balanceBefore := td.GetBalance(stage.creator)
+
+		params := adt.Empty // Missing params required by AddSigner
+		amtSent := abi.NewTokenAmount(1)
+		result := stage.send(stage.msAddr, amtSent, builtin.MethodsMultisig.AddSigner, params, 1)
+		assert.Equal(t, exitcode.Ok, result.Receipt.ExitCode)
+
+		td.AssertBalance(stage.creator, big.Sub(balanceBefore, result.Receipt.GasUsed.Big()))
+		td.AssertBalance(stage.msAddr, multisigBalance) // No change.
+		assert.Equal(t, 1, len(stage.state().Signers))  // No new signers
+	})
+
+	t.Run("fail mismatched params", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+
+		stage := prepareStage(td, acctDefaultBalance, multisigBalance)
+		balanceBefore := td.GetBalance(stage.creator)
+
+		// Wrong params for AddSigner
+		params := multisig.ProposeParams{
+			To:     stage.creator,
+			Value:  big.Zero(),
+			Method: builtin.MethodSend,
+			Params: nil,
+		}
+		amtSent := abi.NewTokenAmount(1)
+		result := stage.send(stage.msAddr, amtSent, builtin.MethodsMultisig.AddSigner, &params, 1)
+		assert.Equal(t, exitcode.Ok, result.Receipt.ExitCode)
+
+		td.AssertBalance(stage.creator, big.Sub(balanceBefore, result.Receipt.GasUsed.Big()))
+		td.AssertBalance(stage.msAddr, multisigBalance) // No change.
+		assert.Equal(t, 1, len(stage.state().Signers))  // No new signers
+	})
+
+	t.Run("fail inner abort", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+
+		stage := prepareStage(td, acctDefaultBalance, multisigBalance)
+		prevHead := td.GetHead(builtin.RewardActorAddr)
+
+		// AwardBlockReward will abort unless invoked by the system actor
+		params := reward.AwardBlockRewardParams{
+			Miner:       stage.creator,
+			Penalty:     big.Zero(),
+			GasReward:   big.Zero(),
+			TicketCount: 100,
+		}
+		amtSent := abi.NewTokenAmount(1)
+		result := stage.send(builtin.RewardActorAddr, amtSent, builtin.MethodsReward.AwardBlockReward, &params, 1)
+		assert.Equal(t, exitcode.Ok, result.Receipt.ExitCode)
+
+		td.AssertBalance(stage.msAddr, multisigBalance) // No change.
+		td.AssertHead(builtin.RewardActorAddr, prevHead)
+	})
+
+	t.Run("fail aborted exec", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+
+		stage := prepareStage(td, acctDefaultBalance, multisigBalance)
+		prevHead := td.GetHead(builtin.InitActorAddr)
+
+		// Illegal paych constructor params (addresses are not accounts)
+		ctorParams := paych.ConstructorParams{
+			From: builtin.SystemActorAddr,
+			To:   builtin.SystemActorAddr,
+		}
+		execParams := init_.ExecParams{
+			CodeCID:           builtin.PaymentChannelActorCodeID,
+			ConstructorParams: chain.MustSerialize(&ctorParams),
+		}
+
+		amtSent := abi.NewTokenAmount(1)
+		result := stage.send(builtin.InitActorAddr, amtSent, builtin.MethodsInit.Exec, &execParams, 1)
```

Comment:
nit: `approverNonce` is constant throughout these tests (`1`), could they instead use a variable that indicates what `1` is?